### PR TITLE
Add API code and types for the compare API

### LINKED ISSUE #358: Add API logic to fetch a diff
Supports #111

### PR DESCRIPTION
**Pull request by willdurand**
Fixes #358

---

This patch adds some code to call the Compare API (not released yet) and some types.

### CODE REVIEW

**Comment by willdurand** [comment]
I updated a type and added some more documentation for the TS synrax.

File: src/api/index.tsx [hunk 1]
```diff
@@ -151,3 +155,25 @@ export const getCurrentUser = async (apiState: ApiState) => {
...
+  addonId,
+  apiState,
+  baseVersionId,
+  headVersionId,
+  path,
+}: GetDiffParams) => {
+  return callApi<ExternalVersionWithDiff>({
+    apiState,
+    endpoint: `reviewers/addon/${addonId}/versions/${baseVersionId}/compare_to/${headVersionId}`,
+    query: path ? { file: path } : undefined,
```
Comment (bobsilverberg) [inline src/api/index.tsx]: When would we not have a path? I guess this is the default screen, where no file has been chosen yet (like with the Browse page)?

File: src/reducers/versions.tsx [hunk 2]
```diff
@@ -80,6 +80,55 @@ export type ExternalVersion = {
...
+  old_path: string;
+  parent: string;
+  path: string;
+  size: number;
+};
+
+// An `ExternalVersionFileWithDiff` object is very similar to an
+// `ExternalVersionFile` object: instead of having a `content` field, it has a
+// `diff` field.
+export type ExternalVersionFileWithDiff = Pick<
```
Comment (bobsilverberg) [inline src/reducers/versions.tsx]: I've not seen this `Pick` syntax before. It's pretty cool. 👍

Comment (bobsilverberg) [inline src/reducers/versions.tsx]: I wonder about the naming, though. These new types are called `ExternalVersionFileWithDiff` and `ExternalVersionWithDiff`, which means that the existing types are just called `ExternalVersionFile` and `ExternalVersion`. That means we are making an assumption about what these "default" types contain. Should those be renamed to `ExternalVersionFileWithFile` and `ExternalVersionWithFile` to make it more explicit what each type contains?

File: src/api/index.tsx (hunk 1 above)
Comment (willdurand) [reply src/api/index.tsx]: Yep, exactly. This is very similar to the Browse API endpoint. There is a default file.

File: src/reducers/versions.tsx (hunk 2 above)
Comment (willdurand) [reply src/reducers/versions.tsx]: That would make all the version types more consistent.

Comment (willdurand) [reply src/reducers/versions.tsx]: I believe it should be something `ExternalVersionFileWithContent` and `ExternalVersionWithContent` though.

**Pull request merged**